packages/flame/lib/src/components/mixins/hoverable.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:meta/meta.dart';
+
+import '../../../game.dart';
+import '../../game/base_game.dart';
+import '../../gestures/events.dart';
+import '../base_component.dart';
+
+mixin Hoverable on BaseComponent {
+  bool _isHovered = false;
+  bool get isHovered => _isHovered;
```

Comment:
Excellent! That was pretty much what I had in mind.

Reply:
yeah what I didn't know is that we had "final" method in dart for the mutator. it's not compiler safe but at least is annotated so good enough for me